main.c

```diff
@@ -0,0 +1,97 @@
+        return -1;
+    }
+
+    switch(input_source) {
+        case CONSOLE_READ:
+            /* Take input array size from user */
+            printf("Please enter size of input array: ");
+            scanf("%d", &(data_entry->input_array_size));
+
+            /* Take elements sequence array size from user */
+            printf("Please enter size of array that needs to be found as the subset of input array: ");
+            scanf("%d", &(data_entry->elements_sequence_size));
+
+            /* Allocate memory for input and elements sequence array */
+            data_entry->input_array = malloc(sizeof(int) * (data_entry->input_array_size));
+            data_entry->elements_sequence = malloc(sizeof(int) * (data_entry->elements_sequence_size));
+
+            if(data_entry->input_array == NULL || data_entry->elements_sequence == NULL) {
+                printf("Memory allocation for input arrays failed.\n");
+                return -1;
+            }
+
+            /* Take input array elements from user */
+            printf("Please enter %d elements of the input array: ", data_entry->input_array_size);
+            for(index = 0; index < (data_entry->input_array_size); index++) {
+                scanf("%d", &(data_entry->input_array[index]));
+            }
+
+            /* Take elements sequence array that needs to be found as the subset from the user */
+            printf("Please enter %d elements of the elements sequence array: ", data_entry->elements_sequence_size);
+            for(index = 0; index < (data_entry->elements_sequence_size); index++) {
+                scanf("%d", &(data_entry->elements_sequence[index]));
+            }
+
+            /* Find smallest possible subset covering the given elements in sequence from input array */
+            find_smallest_subset(data_entry);
+            free(data_entry->input_array);
+            free(data_entry->elements_sequence);
+            break;
+        case FILE_READ:
+            /* Allocate memory to store file name.
+               File name size restricted to a maximum of 512 characters. */
+            file_name = malloc(sizeof(char) * MAX_FILE_NAME_SIZE);
+
+            if(file_name == NULL) {
+                printf("Memory allocation for array to store file name failed.\n");
+                return -1;
+            }
+            printf("Please enter file name from where input needs to be read: ");
+            /* To ensure that scanf reads no more than (MAX_FILE_NAME_SIZE - 1) characters.
+               -1 since a string has to be null ('\0') terminated. */
+            scanf("%s", file_name);
+
+            if(!read_data_from_file(file_name, data_entry)) {
+                printf("Error reading data from file.\n");
+                return -1;
+            }
+
+            free(file_name);
+            break;
+        default:
+            printf("Invalid input source specified.\n");
+            break;
+    }
```

Comment:
This switch is pretty long. Why not move each case to a different file? Mostly taking input. 
Keep a file named handle_input.c, and have 2 functions (hande_file_input and handle_console_input).
Something like this.